provider/datagen/src/transform/cldr/time_zones/mod.rs

```diff
@@ -45,105 +33,72 @@ macro_rules! impl_resource_provider {
-                        let mut resource: cldr_serde::time_zones::time_zone_names::Resource =
-                            serde_json::from_reader(open_reader(&path)?)
-                                .map_err(|e| DataError::from(e).with_path_context(&path))?;
-                        self.time_zone_names_data.insert(
-                            langid.clone(),
-                            Box::new(
-                                resource
-                                    .main
-                                    .0
-                                    .remove(&langid)
-                                    .expect("CLDR file contains the expected language")
-                                    .dates
-                                    .time_zone_names,
-                            ),
-                        )
-                    };
+                    let resource: &cldr_serde::time_zones::time_zone_names::Resource = self
+                        .source
+                        .get_cldr_paths()?
+                        .cldr_dates("gregorian")
+                        .read_and_parse(&langid, "timeZoneNames.json")?;
+                    let time_zone_names = resource
+                        .main
+                        .0
+                        .get(&langid)
+                        .expect("CLDR file contains the expected language")
+                        .dates
+                        .time_zone_names
+                        .clone();
 
-                    if self.bcp47_tzid_data.read().expect("poison").len() == 0 {
-                        let bcp47_time_zone_path = self
-                            .source
-                            .get_cldr_paths()?
-                            .cldr_bcp47()
-                            .join("bcp47")
-                            .join("timezone.json");
+                    let resource: &cldr_serde::time_zones::bcp47_tzid::Resource = self
+                        .source
+                        .get_cldr_paths()?
+                        .cldr_bcp47()
+                        .read_and_parse("timezone.json")?;
 
-                        let resource: cldr_serde::time_zones::bcp47_tzid::Resource =
-                            serde_json::from_reader(open_reader(&bcp47_time_zone_path)?)
-                                .map_err(|e| DataError::from(e).with_path_context(&bcp47_time_zone_path))?;
-                        let r = resource.keyword.u.time_zones.values;
-
-                        let mut data_guard = self.bcp47_tzid_data.write().expect("poison");
-                        for (bcp47_tzid, bcp47_tzid_data) in r.iter() {
-                            if let Some(alias) = &bcp47_tzid_data.alias {
-                                for data_value in alias.split(" ") {
-                                    data_guard.insert(data_value.to_string(), *bcp47_tzid);
-                                }
+                    let mut bcp47_tzids = LiteMap::new();
```

Comment:
Thought: Before, we cached the LiteMap here, but now we have to re-compute the LiteMap each time. Unsure if there is a measurable perf impact. It's probably fine. But if we need to, we could re-introduce the `RwLock` in this specific case.